atlascope/core/job_types/nucleus_to_gland_distance.py

```diff
@@ -14,70 +17,108 @@
 schema = {
     "type": "object",
     "required": [],
-    "properties": {
-        "gland_map_path": {
-            "type": 'string',
-            "title": 'Path to gland map',
-            "default": 'atlascope/core/management/populate/inputs/gland_map.jpg',
-        }
-    },
+    "properties": {},
 }
 
 
+def generate_gland_map(input_image):
+    # https://stackoverflow.com/questions/71865493/is-it-possible-to-create-a-random-shape-on-an-image-in-python
+    img = np.zeros((input_image.shape[0], input_image.shape[1], 3)).astype(np.uint8)
+    rng = np.random.default_rng()
+    noise = rng.integers(0, 255, input_image.shape[:2], np.uint8, True)
+    blur = cv2.GaussianBlur(noise, (0, 0), sigmaX=15, sigmaY=15, borderType=cv2.BORDER_DEFAULT)
+    stretch = skimage.exposure.rescale_intensity(blur, in_range="image", out_range=(0, 255)).astype(
+        np.uint8
+    )
+    thresh = cv2.threshold(stretch, 175, 255, cv2.THRESH_BINARY)[1]
+    kernel = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (9, 9))
+    mask = cv2.morphologyEx(thresh, cv2.MORPH_OPEN, kernel)
+    mask = cv2.morphologyEx(mask, cv2.MORPH_CLOSE, kernel)
+    mask = cv2.merge([mask, mask, mask])
+    result_rgb = cv2.add(img, mask)
+    return np.mean(result_rgb, axis=2).astype(np.uint8)
+
+
+def get_result_image(input_image, nucleus_mask, gland_mask, distances_mask):
+    mask_shape = (input_image.shape[1], input_image.shape[0])
+    nucleus_mask_color = PIL.Image.new("RGBA", mask_shape, "#0ead38")
+    gland_mask_color = PIL.Image.new("RGBA", mask_shape, "#c21d1d")
+    distances_mask_color = PIL.Image.new("RGBA", mask_shape, "#ebe72d")
+
+    original = PIL.Image.fromarray(input_image, mode="L").convert("RGBA")
+    mask_1 = PIL.Image.fromarray(nucleus_mask, mode="L")
+    mask_2 = PIL.Image.fromarray(gland_mask, mode="L")
+    distances_mask = distances_mask.convert("L")
+    composite = PIL.Image.composite(nucleus_mask_color, original, mask_1)
+    composite = PIL.Image.composite(gland_mask_color, composite, mask_2)
+    composite = PIL.Image.composite(distances_mask_color, composite, distances_mask)
+    return composite
+
+
 @shared_task
-def run(job_id: str, original_dataset_id: str, gland_map_path: str):
+def run(job_id: str, original_dataset_id: str):
+    # suppress warnings from high numbers and NaNs in computations
+    # we only need this when we are using a randomly generated gland map
+    import warnings
+
     from atlascope.core.models import Job
 
+    # TODO: remove this warning suppression
+    warnings.filterwarnings("ignore")
```

Comment:
This seems like a bad idea. Better could be to constrain the random mask generation not to cause these warnings in the first place (if possible).